imagebuildah/stage_executor.go

```diff
@@ -985,10 +986,16 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 	}
 	// logCachePulled produces build log for cases when `--cache-from`
 	// is used and a valid intermediate image is pulled from remote source.
-	logCachePulled := func(cacheKey string, remote reference.Named) {
+	logCachePulled := func(cacheKey string, remote types.ImageReference) {
 		if !s.executor.quiet {
 			cachePullMessage := "--> Cache pulled from remote"
-			fmt.Fprintf(s.executor.out, "%s %s\n", cachePullMessage, fmt.Sprintf("%s:%s", remote.String(), cacheKey))
+			repo := ""
+			if remote.Transport().Name() == imagedocker.Transport.Name() {
+				repo = remote.DockerReference().String()
+			} else {
+				repo = remote.StringWithinTransport()
```

Comment:
If this is supposed to represent arbitrary transports, please use `transports.ImageName()`, not a single-transport-scoped name.